File: Calculator/android-native/app/src/main/java/com/qaantum/calculatorhub/calculators/NewConverters.kt
package com.qaantum.calculatorhub.calculators

// NEW CONVERTERS

// Temperature Converter
data class TemperatureResult(val celsius: Double, val fahrenheit: Double, val kelvin: Double)
class TemperatureConverter {
    fun fromCelsius(c: Double) = TemperatureResult(c, c * 9/5 + 32, c + 273.15)
    fun fromFahrenheit(f: Double): TemperatureResult { val c = (f - 32) * 5/9; return TemperatureResult(c, f, c + 273.15) }
    fun fromKelvin(k: Double): TemperatureResult { val c = k - 273.15; return TemperatureResult(c, c * 9/5 + 32, k) }
}

// Length Converter
class LengthConverter {
    private val toMeters = mapOf(
        "Millimeters" to 0.001, "Centimeters" to 0.01, "Meters" to 1.0, "Kilometers" to 1000.0,
        "Inches" to 0.0254, "Feet" to 0.3048, "Yards" to 0.9144, "Miles" to 1609.344, "Nautical Miles" to 1852.0
    )
    
    fun convert(value: Double, from: String, to: String): Double {
        val meters = value * (toMeters[from] ?: 1.0)
        return meters / (toMeters[to] ?: 1.0)
    }
    
    fun getUnits() = toMeters.keys.toList()
}

// Weight Converter
class WeightConverter {
    private val toKg = mapOf(
        "Milligrams" to 0.000001, "Grams" to 0.001, "Kilograms" to 1.0, "Metric Tons" to 1000.0,
        "Ounces" to 0.0283495, "Pounds" to 0.453592, "Stones" to 6.35029, "US Tons" to 907.185
    )
    
    fun convert(value: Double, from: String, to: String): Double {
        val kg = value * (toKg[from] ?: 1.0)
        return kg / (toKg[to] ?: 1.0)
    }
    
    fun getUnits() = toKg.keys.toList()
}

// Volume Converter
class VolumeConverter {
    private val toLiters = mapOf(
        "Milliliters" to 0.001, "Liters" to 1.0, "Cubic Meters" to 1000.0,
        "Teaspoons" to 0.00492892, "Tablespoons" to 0.0147868, "Fluid Ounces (US)" to 0.0295735,
        "Cups (US)" to 0.236588, "Pints (US)" to 0.473176, "Quarts (US)" to 0.946353,
        "Gallons (US)" to 3.78541, "Gallons (UK)" to 4.54609
    )
    
    fun convert(value: Double, from: String, to: String): Double {
        val liters = value * (toLiters[from] ?: 1.0)
        return liters / (toLiters[to] ?: 1.0)
    }
    
    fun getUnits() = toLiters.keys.toList()
}

File: Calculator/ios-native/CalculatorHub/Calculators/NewConverters.swift
import Foundation

// NEW CONVERTERS

// Temperature Converter
struct TemperatureResult { let celsius, fahrenheit, kelvin: Double }
class TemperatureConverter {
    func fromCelsius(_ c: Double) -> TemperatureResult { TemperatureResult(celsius: c, fahrenheit: c * 9/5 + 32, kelvin: c + 273.15) }
    func fromFahrenheit(_ f: Double) -> TemperatureResult { let c = (f - 32) * 5/9; return TemperatureResult(celsius: c, fahrenheit: f, kelvin: c + 273.15) }
    func fromKelvin(_ k: Double) -> TemperatureResult { let c = k - 273.15; return TemperatureResult(celsius: c, fahrenheit: c * 9/5 + 32, kelvin: k) }
}

// Length Converter
class LengthConverter {
    private let toMeters: [String: Double] = ["Millimeters": 0.001, "Centimeters": 0.01, "Meters": 1.0, "Kilometers": 1000.0, "Inches": 0.0254, "Feet": 0.3048, "Yards": 0.9144, "Miles": 1609.344, "Nautical Miles": 1852.0]
    func convert(_ value: Double, from: String, to: String) -> Double { value * (toMeters[from] ?? 1.0) / (toMeters[to] ?? 1.0) }
    func getUnits() -> [String] { Array(toMeters.keys).sorted() }
}

// Weight Converter
class WeightConverter {
    private let toKg: [String: Double] = ["Milligrams": 0.000001, "Grams": 0.001, "Kilograms": 1.0, "Metric Tons": 1000.0, "Ounces": 0.0283495, "Pounds": 0.453592, "Stones": 6.35029, "US Tons": 907.185]
    func convert(_ value: Double, from: String, to: String) -> Double { value * (toKg[from] ?? 1.0) / (toKg[to] ?? 1.0) }
    func getUnits() -> [String] { Array(toKg.keys).sorted() }
}

// Volume Converter
class VolumeConverter {
    private let toLiters: [String: Double] = ["Milliliters": 0.001, "Liters": 1.0, "Cubic Meters": 1000.0, "Teaspoons": 0.00492892, "Tablespoons": 0.0147868, "Fluid Ounces (US)": 0.0295735, "Cups (US)": 0.236588, "Pints (US)": 0.473176, "Quarts (US)": 0.946353, "Gallons (US)": 3.78541, "Gallons (UK)": 4.54609]
    func convert(_ value: Double, from: String, to: String) -> Double { value * (toLiters[from] ?? 1.0) / (toLiters[to] ?? 1.0) }
    func getUnits() -> [String] { Array(toLiters.keys).sorted() }
}
